code/iaas/agent-instance/src/main/java/io/cattle/platform/agent/instance/service/impl/ServiceConsumeMapCreateNicLookup.java

```diff
@@ -30,12 +33,31 @@
         ServiceConsumeMap consumeMap = (ServiceConsumeMap) obj;
         Service service = objectManager.loadResource(Service.class, consumeMap.getServiceId());
         
-        List<? extends ServiceExposeMap> exposeMaps = mapDao.findNonRemoved(ServiceExposeMap.class, Service.class,
-                service.getId());
-        if (exposeMaps.isEmpty()) {
-            return null;
+        // hardcoding the value to mvn projects cross reference
+        if (service.getKind().equalsIgnoreCase("dnsService")) {
+            // get the nics of the instances of the services consuming the DNS service
+            return create().
+                    select(NIC.fields()).
+                    from(NIC)
+                    .join(INSTANCE)
+                    .on(INSTANCE.ID.eq(NIC.INSTANCE_ID))
+                    .join(SERVICE_EXPOSE_MAP)
+                    .on(SERVICE_EXPOSE_MAP.INSTANCE_ID.eq(INSTANCE.ID))
+                    .join(SERVICE_CONSUME_MAP)
+                    .on(SERVICE_CONSUME_MAP.SERVICE_ID.eq(SERVICE_EXPOSE_MAP.SERVICE_ID))
+                    .where(SERVICE_CONSUME_MAP.CONSUMED_SERVICE_ID.eq(service.getId()))
+                    .fetchInto(NicRecord.class);
```

Comment:
Same comment here.  put some minimal is null check somewhere to prevent forever growing lists.